swap/htlc.go

```diff
@@ -171,24 +171,27 @@ func NewHtlc(version ScriptVersion, cltvExpiry int32,
 		// the p2wsh witness program corresponding to the matching
 		// public key of this address.
 		sigScript, err = txscript.NewScriptBuilder().
-			AddData(p2wshPkScript).
+			AddData(pkScript).
 			Script()
 		if err != nil {
 			return nil, err
 		}
 
 		address, err = btcutil.NewAddressScriptHash(
-			p2wshPkScript, chainParams,
+			nestedPkScript, chainParams,
```

Comment:
This is the wrong script.

Reply:
I assume this would've been caught by the server side itest. Makes me think we should have a server side PR that references this PR to make sure everything still passes.

Reply:
Argh missed this... Great catch @guggero !